sql/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -105,6 +111,64 @@ public FunctionInfo info() {
         }
     }
 
+    private static class IpEqOperator extends Operator<Object> {
+
+        private final FunctionInfo info;
+
+        IpEqOperator(FunctionInfo info) {
+            this.info = info;
+        }
+
+        @Override
+        @SafeVarargs
+        public final Boolean evaluate(TransactionContext txnCtx, Input<Object>... args) {
+            assert args.length == 2 : "number of args must be 2";
+            String left = extractIpLiteral(args[0]);
+            if (null == left) {
+                return null;
+            }
+            String right = extractIpLiteral(args[1]);
+            if (null == right) {
+                return null;
+            }
+            if (left.contains("/")) {
+                return IpAddressRange.of(left).inRange(right);
+            }
+            if (right.contains("/")) {
+                return IpAddressRange.of(right).inRange(left);
+            }
+            return Objects.equals(left, right);
+        }
+
+        private static String extractIpLiteral(Input<Object> input) {
+            if (input instanceof Literal) {
+                return (String) input.value();
+            }
+
+            if (input instanceof FunctionExpression) {
```

Comment:
We should find a way to do this without having concrete `Input` implementation knowledge. A scalar implementation should always only call `value` on `Input` and not be coupled to concrete implementations.